src/ResourceManager/Network/Commands.Network/VirtualNetwork/Subnet/NewAzureVirtualNetworkSubnetConfigCommand.cs

```diff
@@ -59,6 +60,17 @@ public override void Execute()
                 subnet.RouteTable.Id = this.RouteTableId;
             }
 
+            if (this.PrivateAccessService != null)
+            {
+                subnet.PrivateAccessServices = new List<PSPrivateAccessService>();
+                foreach (var item in this.PrivateAccessService)
+                {
+                    var service = new PSPrivateAccessService();
+                    service.Service = item;
+                    subnet.PrivateAccessServices.Add(service);
+                }
```

Comment:
@EvgenyAgafonchikov in the class below (`SetAzureVirtualNetworkSubnetConfigCommand.cs`), you have an else that sets the `subnet.PrivateAccessServices` to null. Should the same be done in this class?